adafruit_tlc59711.py

```diff
@@ -542,44 +556,44 @@ def calculate_BCData(*, Ioclmax=18, IoutR=17, IoutG=15, IoutB=9):
         # Iout / Ioclmax       =  BCX / 127              | * 127
         # Iout / Ioclmax * 127 =  BCX
         if not 2.0 <= Ioclmax <= 60.0:
-            raise ValueError("Ioclmax {} not in range: 2mA..60mA" "".format(Ioclmax))
+            raise ValueError(f"Ioclmax {Ioclmax} not in range: 2mA..60mA")
         if not 0.0 <= IoutR <= Ioclmax:
-            raise ValueError("IoutR {} not in range: 2mA..{}mA".format(IoutR, Ioclmax))
+            raise ValueError(f"IoutR {IoutR} not in range: 2mA..{Ioclmax}mA")
         if not 0.0 <= IoutG <= Ioclmax:
-            raise ValueError("IoutG {} not in range: 2mA..{}mA".format(IoutG, Ioclmax))
+            raise ValueError(f"IoutG {IoutG} not in range: 2mA..{Ioclmax}mA")
         if not 0.0 <= IoutB <= Ioclmax:
-            raise ValueError("IoutB {} not in range: 2mA..{}mA".format(IoutB, Ioclmax))
+            raise ValueError(f"IoutB {IoutB} not in range: 2mA..{Ioclmax}mA")
         bcr = int((IoutR / Ioclmax) * 127)
         bcg = int((IoutG / Ioclmax) * 127)
         bcb = int((IoutB / Ioclmax) * 127)
         if not 0 <= bcr <= 127:
-            raise ValueError("bcr {} not in range: 0..127".format(bcr))
+            raise ValueError(f"bcr {bcr} not in range: 0..127")
         if not 0 <= bcg <= 127:
-            raise ValueError("bcg {} not in range: 0..127".format(bcg))
+            raise ValueError(f"bcg {bcg} not in range: 0..127")
         if not 0 <= bcb <= 127:
-            raise ValueError("bcb {} not in range: 0..127".format(bcb))
+            raise ValueError(f"bcb {bcb} not in range: 0..127")
         return (bcr, bcg, bcb)
 
     ##########################################
 
     @staticmethod
-    def _convert_01_float_to_16bit_integer(value):
+    def _convert_01_float_to_16bit_integer(value: float) -> int:
         """Convert 0..1 Float Value to 16bit (0..65535) Range."""
         # check if value is in range
-        if not 0.0 <= value[0] <= 1.0:
```

Comment:
It seems like maybe this was never used before? It looks like it would raise an exception error, the only place that calls this function verifies the type of `value` as a float before calling in to here so it could never allow this bracketed index access AFAIK.